src/node_quirc_decode.c

```diff
@@ -415,3 +421,49 @@ nq_load_jpeg(struct quirc *q, const uint8_t *img, size_t img_len)
 	jpeg_destroy_decompress(&dinfo);
 	return -1;
 }
+
+static int
+nq_load_raw(struct quirc *q, const uint8_t *img, size_t img_len, size_t width, size_t height)
+{
+	if (quirc_resize(q, width, height) < 0)
+		goto fail;
+
+	uint8_t *image = quirc_begin(q, NULL, NULL);
+
+	int channels = img_len / width / height;
+
+	if (channels == 1)
+	{
+		memcpy(image, img, img_len);
+	}
+	else if (channels == 3 || channels == 4)
+	{
+		size_t dst_offset = width * height - 1;
+		size_t src_offset = dst_offset * channels;
+		do
+		{
+			uint8_t r = img[src_offset];
+			uint8_t g = img[src_offset + 1];
+			uint8_t b = img[src_offset + 2];
+			// ignore alpha, if present
+			image[dst_offset] = (uint8_t)(0.2126 * (float)r + 0.7152 * (float)g + 0.0722 * (float)b);
```

Comment:
Theses constants come from https://en.wikipedia.org/wiki/Grayscale#Colorimetric_(perceptual_luminance-preserving)_conversion_to_grayscale correct? If yes, can we add a comment with this link for documentation purpose please?

Reply:
👍🏻